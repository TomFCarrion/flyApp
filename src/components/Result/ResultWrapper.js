import React from 'react';


const ResultWrapper = () => {
    
    return (
        <h1>Search Your Flight </h1>
    );
};
export default ResultWrapper;

