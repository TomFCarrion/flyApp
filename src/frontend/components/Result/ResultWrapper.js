import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { FETCH_FLIGHTS } from '../../actions/actionTypes';
import Result from './Result';
import axios from 'axios';

const ResultWrapper = () => {
  const data = useSelector((state) => state.flightsReducer);
  const from = useSelector((state) => state.infoReducer.from);
  const fromId = useSelector((state) => state.infoReducer.fromId);
  const to = useSelector((state) => state.infoReducer.to);
  const toId = useSelector((state) => state.infoReducer.toId);
  const startDate = useSelector((state) => state.infoReducer.startDate);
  const endDate = useSelector((state) => state.infoReducer.endDate);
  const dispatch = useDispatch();

  const options = {
    method: 'GET',
    url: `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices/browseroutes/v1.0/US/USD/en-US/${fromId}/${toId}/${startDate}`,
    params: { inboundpartialdate: endDate },
    headers: {
      'x-rapidapi-key': '08eb576b39mshaa3eb3f0249657ep1b15f7jsn598ba72db2e4',
      'x-rapidapi-host':
        'skyscanner-skyscanner-flight-search-v1.p.rapidapi.com',
    },
  };

  function fetchData() {
    return (dispatch) => {
      axios
        .request(options)
        .then((res) =>
          dispatch({
            type: FETCH_FLIGHTS,
            data: res.data,
          }),
        )
        .then((res) => console.error(res))
        .catch((error) => {
          console.error(error);
        });
    };
  }

  const parseData = () => {
    let carriers = [];
    let quotes = [];
    let parsedCarriers = [];

    if (data?.data?.Carriers?.length) {
      carriers = data.data.Carriers;
      quotes = data.data.Quotes;

      for (let i = 0; i < carriers.length; i++) {
        for (let j = 0; j < quotes.length; j++) {
          if (carriers[i].CarrierId === quotes[j].OutboundLeg.CarrierIds[0]) {
            parsedCarriers.push({
              id: carriers[i].CarrierId,
              name: carriers[i].Name,
              price: quotes[j].MinPrice,
            });
          }
        }
      }
    }

    console.log(parsedCarriers);
    return parsedCarriers;
  };

  useEffect(() => {
    dispatch(fetchData());
  }, []);

  return (
    <div>
      <Link to="/">
        <div className="navBack"> </div>
      </Link>

      <ul>
        { parseData().map( ticket => 
          <Result
            key ={ ticket.id }
            from ={ from }
            to = { to }
            airlineName = { ticket.name }
            flightNumber = { ticket.id }
            ticketPrice = { ticket.price }    
            date = { startDate }
          />)
        }
      </ul>
    </div>
  );
};

export default ResultWrapper;
