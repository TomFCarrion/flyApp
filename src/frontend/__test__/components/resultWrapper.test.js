import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ResultWrapper from '../../components/Result/ResultWrapper';
import ProviderMock from '../../__mocks__/providerMock';

describe(' <ResultWrapper/>', () => {
  test('ResultWrapper component render', () => {
    const result = shallow(
      <ProviderMock>
        <ResultWrapper />
      </ProviderMock>,
    );
    expect(result.length).toEqual(1);
  });
});

describe('ResultWrapper snapshot', () => {
  test('Check Result snapshot', () => {
    const searchWrapper = create(
      <ProviderMock>
        <ResultWrapper />
      </ProviderMock>,
    );
    expect(searchWrapper.toJSON()).toMatchSnapshot();
  });
});
