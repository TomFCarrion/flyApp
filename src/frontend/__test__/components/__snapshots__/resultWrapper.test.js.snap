// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultWrapper snapshot Check Result snapshot 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="navBack"
    >
       
    </div>
  </a>
  <ul />
</div>
`;
